tests/unit/s2n_client_psk_extension_test.c

```diff
@@ -160,6 +189,67 @@ int main(int argc, char **argv)
             EXPECT_SUCCESS(s2n_connection_free(conn));
             EXPECT_SUCCESS(s2n_stuffer_free(&out));
         }
+
+        /* On the second ClientHello after a retry request,
+         * do not send any PSKs that do not match the cipher suite. */
+        {
+            struct s2n_stuffer out = { 0 };
+            EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&out, 0));
+
+            const struct s2n_psk_test_case matching_psk = {
+                    .hmac_alg = S2N_HMAC_SHA384, .hash_size = SHA384_DIGEST_LENGTH,
+                    .identity = test_identity, .identity_size = sizeof(test_identity)
+            };
+            const struct s2n_psk_test_case non_matching_psk = {
+                    .hmac_alg = S2N_HMAC_SHA224, .hash_size = SHA224_DIGEST_LENGTH,
+                    .identity = test_identity, .identity_size = sizeof(test_identity)
+            };
+            struct s2n_psk_test_case test_cases[] = { matching_psk, non_matching_psk };
+
+            struct s2n_connection *conn;
+            EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
+            conn->handshake.handshake_type = HELLO_RETRY_REQUEST;
+            conn->secure.cipher_suite = &s2n_tls13_aes_256_gcm_sha384;
+            EXPECT_EQUAL(conn->secure.cipher_suite->prf_alg, matching_psk.hmac_alg);
+
+            for (size_t i = 0; i < s2n_array_len(test_cases); i++) {
+                struct s2n_psk *psk = NULL;
+                EXPECT_OK(s2n_array_pushback(&conn->psk_params.psk_list, (void**) &psk));
+                EXPECT_SUCCESS(s2n_psk_init(psk, S2N_PSK_TYPE_EXTERNAL));
+                EXPECT_SUCCESS(s2n_psk_new_identity(psk, test_cases[i].identity, test_cases[i].identity_size));
+                psk->hmac_alg = test_cases[i].hmac_alg;
+            }
+
+            EXPECT_SUCCESS(s2n_client_psk_extension.send(conn, &out));
+
+            /* The identity list should only contain the matching psk's identity.
+             * It should NOT contain the non-matching psk. */
+
+            uint16_t identity_list_size = 0;
+            uint16_t identity_size = 0;
+            uint8_t *identity_data;
```

Comment:
Nit: uint8_t *identity_data = NULL;

Reply:
Not really necessary (the value of identity_data is set on line 238) but I guess I could do it to match the others.